### PR TITLE
fix(axvisor): correct shell filesystem command handling

### PR DESCRIPTION
**Pull request by YanLien**
## Problem

The Axvisor shell filesystem commands and shutdown path had several compatibility and data consistency issues:

- `touch` used `File::create`, which truncated existing files.
- `mv` fell back to copy-and-delete after any `rename` failure, potentially hiding permission and other errors.
- `rm -f` ignored all metadata errors instead of only missing files.
- `ls` silently discarded directory iteration errors.
- `cp` did not clearly enforce its supported argument layout.
- Axvisor could exit without fully unmounting the host filesystem, leaving ext4 metadata insufficiently persisted.
- When the RTC was not initialized, ext4 `i_dtime` could fall within the valid inode-number range and be mistaken for an orphan-list pointer.
- POSIX `stat` opened the target for reading, preventing metadata queries on files without read permission.
- Some AArch64 open flag values in `fcntl.h` did not match the Linux ABI.
- Bindgen did not always receive the cross-compilation target and could generate bindings using the host ABI.

## Changes

- Corrected Axvisor shell filesystem command behavior:
  - Use `OpenOptions` without truncation for `touch`.
  - Fall back to copy-and-delete in `mv` only for cross-device rename errors.
  - Make `rm -f` ignore only `NotFound`.
  - Propagate directory iteration errors from `ls`.
  - Require exactly one source and one destination for `cp`.
  - Use a shared metadata-based path type check.

- Completed the filesystem shutdown path:
  - Add a `shutdown` operation to `FilesystemOps`, defaulting to `flush`.
  - Perform a proper ext4 unmount during shutdown.
  - Shut down host filesystems and release block-device IRQ registrations before Axvisor exits.
  - Export `shutdown_host_filesystems` from the architecture-independent `axvm` API.

- Corrected ext4 inode deletion metadata:
  - Explicitly clear the inode link count when freeing an inode.
  - Keep the deletion time no earlier than the filesystem's last write time.
  - Ensure `i_dtime` is outside the valid inode-number range.
  - Add unit tests for deletion-time boundary conditions.

- Corrected POSIX and ABI compatibility:
  - Implement `stat` and `lstat` using path metadata without opening the target.
  - Define `O_DIRECT`, `O_DIRECTORY`, `O_LARGEFILE`, and `O_NOFOLLOW` according to the AArch64 Linux ABI.
  - Always pass the cross-compilation target to bindgen.
  - Convert Rust `riscv64gc-*` targets into Clang-compatible `riscv64-*` triples.

## Implementation Rationale

Filesystem commands now perform fallback behavior only for errors where recovery is semantically valid. For example, `mv` uses copy-and-delete only when `rename` fails because the source and destination are on different filesystems. Permission, device, and other errors are returned directly.

The shutdown path now crosses an explicit VFS boundary and performs an ext4 unmount before the backing device or its IRQ registrations become unavailable. This ensures that journal, superblock, and inode state are written back in the correct order.

When the system clock is not initialized, ext4 selects the deletion time from the current time, the filesystem's last write time, and the first value outside the valid inode-number range. This prevents time regression and avoids ambiguity with orphan inode pointers.

### CODE REVIEW

File: os/axvisor/src/shell/command/base.rs [hunk 1]
```diff
@@ -49,54 +47,97 @@ fn split_whitespace(s: &str) -> (&str, &str) {
...
-            let file_type = metadata.file_type();
-            let file_type_char = file_type_to_char(file_type);
-            let rwx = file_perm_to_rwx(metadata.permissions().mode());
-            let rwx = unsafe { core::str::from_utf8_unchecked(&rwx) };
-            println!("{}{} {:>8} {}", file_type_char, rwx, size, entry);
-        } else {
-            println!("{}", entry);
-        }
-        Ok(())
+fn path_is_dir(path: &str) -> io::Result<bool> {
```
Comment (ZR233) [inline os/axvisor/src/shell/command/base.rs:50]: 【阻塞｜缺少回归】这个新判定是 `ls/rm/mv/cp` 修复的共同边界，但 PR 只有实现、没有驱动真实生产函数的测试；旧实现恢复后 CI 仍会通过。请加入确定性回归，构造“目录可 `read_dir` 但 metadata 行为受限”、普通文件、ENOENT 与其他错误，并同时锁定 touch/rename/cp 的修复语义。

**Review by ZR233 — CHANGES_REQUESTED** [review]
实现方向合理，current-head 58 个 CI check 通过；但这是同时修复 `ls/rm/mv/touch/cp` 多个可观察行为的 bugfix，PR 没有增加任何确定性回归测试。按仓库要求，bugfix 需先用会在旧实现上失败的测试锁定行为，至少覆盖目录 metadata 受限时的识别、`rm -f` 仅忽略 ENOENT、非 EXDEV rename 不回退、touch 不截断、cp 多参数拒绝。
